src/treetensornetworks/opsum_to_ttn.jl

```diff
@@ -40,92 +44,79 @@ Hamiltonian, compressing shared interaction channels.
 """
 function ttn_svd(os::OpSum, sites::IndsNetwork, root_vertex; kwargs...)
   # Function barrier to improve type stability
-  coefficient_type = ITensorMPS.determineValType(terms(os))
+  coefficient_type = determine_val_type(terms(os))
   return ttn_svd(coefficient_type, os, sites, root_vertex; kwargs...)
 end
 
-function ttn_svd(
-  coefficient_type::Type{<:Number},
-  os::OpSum,
-  sites0::IndsNetwork,
-  root_vertex;
-  mindim::Int=1,
-  maxdim::Int=typemax(Int),
-  cutoff=eps(real(coefficient_type)) * 10,
-)
-  linkdir_ref = ITensors.In   # safe to always use autofermion default here
+function remove_dups!(v::Vector)
+  N = length(v)
+  (N == 0) && return nothing
+  sort!(v)
+  n = 1
+  u = 2
+  while u <= N
+    while u < N && v[u] == v[n]
+      u += 1
+    end
+    if v[u] != v[n]
+      v[n + 1] = v[u]
+      n += 1
+    end
+    u += 1
+  end
+  resize!(v, n)
+  return nothing
+end
+
+function pos_in_link!(linkmap::Dict, k)
+  isempty(k) && return -1
+  pos = get(linkmap, k, -1)
+  if pos == -1
+    pos = length(linkmap) + 1
+    linkmap[k] = pos
+  end
+  return pos
+end
 
-  sites = copy(sites0)  # copy because of modification to handle internal indices 
+function make_sparse_ttn(
+  os::OpSum, sites::IndsNetwork, root_vertex, coefficient_type, calc_qn
```

Comment:
```suggestion
  coefficient_type::Type, os::OpSum, sites::IndsNetwork; root_vertex, calc_qn
```
Conventionally types are input as the first arguments. Also I think it is clearer if we make `root_vertex` and `calc_qn` keyword arguments.